clmm/dataops.py

```diff
@@ -0,0 +1,275 @@
+
+    In the flat sky approximation, these angles are calculated using
+    .. math::
+        \theta = \sqrt{\left(\delta_s-\delta_l\right)^2+
+        \left(\alpha_l-\alpha_s\right)^2\cos^2(\delta_l)}
+
+        \tan\phi = \frac{\delta_s-\delta_l}{\left(\alpha_l-\alpha_s\right)\cos(\delta_l)}
+
+    For extended descriptions of parameters, see `compute_shear()` documentation.
+    """
+    if not -360. <= ra_lens <= 360.:
+        raise ValueError(f"ra = {ra_lens} of lens if out of domain")
+    if not -90. <= dec_lens <= 90.:
+        raise ValueError(f"dec = {dec_lens} of lens if out of domain")
+    if not all(-360. <= x_ <= 360. for x_ in ra_source_list):
+        raise ValueError("Cluster has an invalid ra in source catalog")
+    if not all(-90. <= x_ <= 90 for x_ in dec_source_list):
+        raise ValueError("Cluster has an invalid dec in the source catalog")
+    # Put angles between -pi and pi
+    r2pi = lambda x: x-np.round(x/(2.0*math.pi))*2.0*math.pi
+    deltax = r2pi(np.radians(ra_source_list-ra_lens))*math.cos(math.radians(dec_lens))
+    deltay = np.radians(dec_source_list-dec_lens)
+    # Ensure that abs(delta ra) < pi
+    #deltax[deltax >= np.pi] = deltax[deltax >= np.pi]-2.*np.pi
+    #deltax[deltax < -np.pi] = deltax[deltax < -np.pi]+2.*np.pi
+    angsep = np.sqrt(deltax**2+deltay**2)
+    phi = np.arctan2(deltay, -deltax)
+    # Forcing phi to be zero everytime angsep is zero. This is necessary due to arctan2 features (it returns ).
+    phi[angsep==0.0] = 0.0
+    if np.any(angsep > np.pi/180.):
+        warnings.warn("Using the flat-sky approximation with separations >1 deg may be inaccurate")
+    return angsep, phi
+def _compute_tangential_shear(shear1, shear2, phi):
+    r"""Compute the tangential shear given the two shears and azimuthal positions for
+    a single source or list of sources.
+
+    We compute the tangential shear following Eq. 7 of Schrabback et al. 2018, arXiv:1611:03866
+    .. math::
+        g_t = -\left( g_1\cos\left(2\phi\right)-g_2\sin\left(2\phi\right)\right)
+
+    For extended descriptions of parameters, see `compute_shear()` documentation.
+    """
+    return -(shear1*np.cos(2.*phi)+shear2*np.sin(2.*phi))
+def _compute_cross_shear(shear1, shear2, phi):
+    r"""Compute the cross shear given the two shears and azimuthal position for a single
+    source of list of sources.
+
+    We compute the cross shear following Eq. 8 of Schrabback et al. 2018, arXiv:1611:03866
+    also checked arxiv 0509252
+    .. math::
+        g_x = g_1 \sin\left(2\phi\right)-g_2\cos\left(2\phi\right)
+
+    For extended descriptions of parameters, see `compute_shear()` documentation.
+    """
+    return shear1*np.sin(2.*phi)-shear2*np.cos(2.*phi)
+def make_transversal_profile(components, angsep, angsep_units, bin_units,
+                        bins=10, include_empty_bins=False,
+                        return_binnumber=False,
+                        cosmo=None, z_lens=None):
```

Comment:
This used to be called `make_binned_profile` and I am not sure I understand the need for a different name. Could you please clarify your thoughts there. I find the adjective `transversal` quite confusing and I don't think it is used in the literature. I'd suggest `radial` instead and maybe `make_binned_radial_profile` for the function name, which is very explicit (but maybe too long). Pleas share your thoughts :) .

Reply:
@combet I was not very happy with the name `make_binned_profile`, it sounded redundant and not explicit enough at the same time. I added the word `transversal` because that is how we were binning, in a transversal distance (be it angular or physical). But I understand it is not widely used and it might generate confusion. How about `make_radial_profile`?

Reply:
@m-aguena , yes, `make_radial_profile` would work well, with the radial distance that can be understood as angular or physical distance.

Reply:
done!